crates/base/src/worker_ctx.rs

```diff
@@ -112,18 +127,39 @@ pub async fn create_user_worker_pool() -> Result<mpsc::UnboundedSender<UserWorke
 
     let user_worker_msgs_tx_clone = user_worker_msgs_tx.clone();
     let _handle: tokio::task::JoinHandle<Result<(), Error>> = tokio::spawn(async move {
-        let mut user_workers: HashMap<Uuid, mpsc::UnboundedSender<WorkerRequestMsg>> =
+        let mut user_workers: HashMap<u64, mpsc::UnboundedSender<WorkerRequestMsg>> =
             HashMap::new();
 
         loop {
             match user_worker_msgs_rx.recv().await {
                 None => break,
                 Some(UserWorkerMsgs::Create(mut worker_options, tx)) => {
-                    let key = Uuid::new_v4();
                     let mut user_worker_rt_opts = match worker_options.conf {
                         EdgeContextOpts::UserWorker(opts) => opts,
                         _ => unreachable!(),
                     };
+
+                    // derive worker key from service path
+                    // if force create is set, add current epoch mili seconds to randomize
+                    let service_path = worker_options.service_path.to_str().unwrap_or("");
+                    let mut key_input = service_path.to_string();
+                    if user_worker_rt_opts.force_create {
+                        let cur_epoch_time = SystemTime::now().duration_since(UNIX_EPOCH)?;
+                        key_input = format!("{}-{}", key_input, cur_epoch_time.as_millis());
+                    }
+                    let key = city_hash_64(key_input.as_bytes());
+
+                    // do not recreate the worker if it already exists
+                    if let Some(_worker) = user_workers.get(&key) {
```

Comment:
Why this check if `force_create` is enabled?

Reply:
ah yeah, good point. Probably better to swap the two checks